test/gpu/fuse_mlir.cpp

```diff
@@ -452,6 +455,191 @@ TEST_CASE(int_quant_dot_tanh_fails)
     EXPECT(has_pointwise);
 }
 
+TEST_CASE(conv_split_reduce)
+{
+    migraphx::shape s_x{migraphx::shape::float_type, {2, 4, 64, 64}};
+    migraphx::shape s_w{migraphx::shape::float_type, {320, 4, 3, 3}};
+    migraphx::shape s_b{migraphx::shape::float_type, {32}};
+    migraphx::program p1;
+    {
+        auto* mm = p1.get_main_module();
+        auto x   = mm->add_parameter("x", s_x);
+        auto w   = mm->add_parameter("w", s_w);
+        auto b   = mm->add_literal(migraphx::generate_literal(s_b));
+        auto mb  = mm->add_instruction(
+            migraphx::make_op("broadcast", {{"axis", 1}, {"out_lens", {2, 32, 10, 64, 64}}}), b);
+        auto conv = mm->add_instruction(
+            migraphx::make_op("convolution", {{"padding", {1, 1, 1, 1}}}), x, w);
+        auto reshape = mm->add_instruction(
+            migraphx::make_op("reshape", {{"dims", {2, 32, 10, 64, 64}}}), conv);
+        auto add = add_pointwise(p1, "main:pointwise0", {reshape, mb}, single_pointwise("add"));
+        auto mean_var = add_reduce(
+            p1,
+            "main:split_reduce0",
+            {add},
+            {2, 3, 4},
+            "assign_add",
+            [&](auto* rm,
+                const auto& inputs,
+                const auto& axes) -> std::vector<migraphx::instruction_ref> {
+                auto xx    = add_pointwise(p1, rm, "main:pointwise1", {inputs[0]}, squared());
+                auto rsum1 = rm->add_instruction(migraphx::make_op("reduce_sum", {{"axes", axes}}),
+                                                 inputs[0]);
+                auto rsum2 =
+                    rm->add_instruction(migraphx::make_op("reduce_sum", {{"axes", axes}}), xx);
+                return {rsum2, rsum1};
+            });
+        auto var =
+            mm->add_instruction(migraphx::make_op("get_tuple_elem", {{"index", 0}}), mean_var);
+        auto mean =
+            mm->add_instruction(migraphx::make_op("get_tuple_elem", {{"index", 1}}), mean_var);
+        mm->add_return({var, mean});
+    }
+    run_pass(p1);
+    migraphx::program p2;
+    {
+        auto* mm = p2.get_main_module();
+        auto x   = mm->add_parameter("x", s_x);
+        auto w   = mm->add_parameter("w", s_w);
+        auto b   = mm->add_literal(migraphx::generate_literal(s_b));
+        auto mb  = mm->add_instruction(
+            migraphx::make_op("broadcast", {{"axis", 1}, {"out_lens", {2, 32, 10, 64, 64}}}), b);
+        auto fused =
+            add_mlir(p2,
+                     "mlir_main:pointwise0_main:split_reduce0",
```

Comment:
minor: why do we lose "convolution" in the name the MLIR instruction?

Reply:
Names are constructed from modules that are fused.  convolution or dot would appear as `mlir_op[op=""` attribute. 
https://github.com/ROCm/AMDMIGraphX/blob/05b2ff452ae8585ed1d191d907e7e95c448bc318/src/targets/gpu/fuse_mlir.cpp#L140